server.js

```diff
@@ -1,30 +1,197 @@
+      message: "OK",
+      body: {
+        company: company
+      }
+    });
+  }
+});
+
+
+// Use for example http://localhost:8081/companies/states/california to get data for all companies with a HQ in California
+// https://project-express-api-cvzekbgn3q-lz.a.run.app/companies/states/california
+app.get('/companies/states/:state', (req, res) => {
+  const states = decodeURIComponent(req.params.state).toLowerCase();
+  const showSectors = req.query.sectors;
+  let companiesFromStates = data.filter((item) => item.hq_state.toLowerCase() === states);
+  
+  if (showSectors) {
+    companiesFromStates = companiesFromStates.filter((item) => item.sectors);
+  }
+  
+  if (companiesFromStates.length === 0) {
+    res.status(200).json({
+      success: false,
+      message: `None of the top 50 companies has their HQ in: ${states}`,
+      body: {}
+    });
+  } else {
+    res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        companies: companiesFromStates
+      }
+    });
+  }
 });
 
+
+// Use for example http://localhost:8081/companies/years/2004 to get data for all companies founded in 2004
+// https://project-express-api-cvzekbgn3q-lz.a.run.app/companies/years/2004
+app.get('/companies/years/:year', (req, res) => {
+  const years = req.params.year;
+  const showYears = req.query.founding_year;
+  let companiesFoundedYears = data.filter((item) => item.founding_year === +years);
+  
+  if (showYears) {
+    companiesFoundedYears = companiesFoundedYears.filter((item) => item.years);
+  }
+  
+  if (companiesFoundedYears.length === 0) {
+    res.status(200).json({
+      success: false,
+      message: `None of the top 50 companies were founded in the year ${years}`,
+      body: {}
+    });
+  } else {
+    res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        companies: companiesFoundedYears
+      }
+    });
+  }
+});
+
+// Use for example http://localhost:8081/companies/sectors/software%20infrastructure/california to get data for companies
+// in the software infrastructure sector with a HQ in California
+// https://project-express-api-cvzekbgn3q-lz.a.run.app/companies/sectors/software%20infrastructure/california
+app.get('/companies/sectors/:sector/:state', (req, res) => {
+  const sectors = decodeURIComponent(req.params.sector).toLowerCase();
+  const state = req.params.state.toLowerCase();
+  let companiesInSectorsInStates = data.filter(
+    (item) =>
+      item.sector.toLowerCase() === sectors &&
+      item.hq_state.toLowerCase() === state
+  );
+
+  if (companiesInSectorsInStates.length === 0) {
+    res.status(200).json({
+      success: false,
+      message: `None of the top 50 companies match the provided ${sectors} sector and ${state} state criteria`,
+      body: {}
+    });
+  } else {
+    res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        companies: companiesInSectorsInStates
+      }
+    });
+  }
+});
```

Comment:
Great job using 2 params as endpoints and then filtering them out. This is getting complex...